### PR TITLE
[UF-508] Layout Editor should Support Page Layouts

### PR DESCRIPTION
**Pull request by ederign**
Added support on layout editor for 100% height page layouts.

We now have to layout modes: FLUID and PAGE (100% height).

Also, I've added a new drag component called TargetDivDragComponent: a new dnd component that renders only a div + id (will be used by page building stuff).

image

### CODE REVIEW

**Comment by ederign** [comment]
Tomorrow I'll record a video for this feature.

**Comment by tomasdavidorg** [comment]
Jenkins execute full downstream build

**Comment by psiroky** [comment]
Jenkins execute full downstream build

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/components/container/Container.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import javax.enterprise.inject.Instance;
 import javax.inject.Inject;
 
+import com.google.gwt.core.client.GWT;
```
Comment (pefernan) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/components/container/Container.java]: @ederign Unused import

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/components/container/ContainerView.java [hunk 2]
```diff
@@ -85,4 +85,10 @@ public void addEmptyRow(UberElement<EmptyDropRow> emptyDropRow) {
                     "container-empty");
```
Comment (pefernan) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/components/container/ContainerView.java]: Some unused imports here too

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/components/rows/Row.java [hunk 3]
```diff
@@ -26,10 +26,12 @@
 import javax.enterprise.inject.Instance;
 import javax.inject.Inject;
 
+import com.google.gwt.core.client.GWT;
```
Comment (pefernan) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/components/rows/Row.java]: Some unused imports here too.

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/components/rows/RowView.java [hunk 4]
```diff
@@ -19,6 +19,7 @@
 import javax.enterprise.context.Dependent;
 import javax.inject.Inject;
 
+import com.google.gwt.core.client.GWT;
```
Comment (pefernan) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/components/rows/RowView.java]: Unused import

File: uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorPresenter.java [hunk 5]
```diff
@@ -24,6 +24,7 @@
 import javax.enterprise.event.Event;
 import javax.inject.Inject;
 
+import com.google.gwt.core.client.GWT;
```
Comment (pefernan) [inline uberfire-extensions/uberfire-layout-editor/uberfire-layout-editor-client/src/main/java/org/uberfire/ext/layout/editor/client/LayoutEditorPresenter.java]: Unused import.

**Comment by ederign** [comment]
@pefernan thanks for the review. Don't need to test this on master. I'll fix it on this PR anyway.

**Comment by ederign** [comment]
@pefernan changes done. Thanks for the review. I fixed the first two issues, but I cannot reproduce the third one (maybe only happens on forms). 
Could you please create a JIRA and I'll take a look on this on the next sprint?

**Comment by tomasdavidorg** [comment]
Jenkins execute full downstream build

**Comment by ederign** [comment]
the failing test it's not related: org.jbpm.test.functional.SplitNodeAfterStartTest.testSplitNodeAfterStart

**Comment by ederign** [comment]
@paulovmr fixed!

**Comment by ederign** [comment]
jenkins retest this

**Comment by tomasdavidorg** [comment]
Hi @ederign, so far I found two issues. First one is when you add two html components, move the second one to the same row (under or above) to create the "box" and delete both components. You will get this modal dialog.
screenshot from 2017-06-06 14-48-27

The second problem is when you add a screen component, The "edit/remove" button (three dots) is hardly accessible to click because it is behind the content.

Tomorrow I going to test it deeply and i will create jiras for the issues.

**Comment by ederign** [comment]
@tomasdavidorg the dots are fixed now. I'll take a look at the other issue asap. Thanks for the review.

**Comment by ederign** [comment]
https://issues.jboss.org/browse/UF-530

**Pull request merged**